pkgs/tools/security/bitwarden/default.nix

```diff
@@ -1,73 +1,117 @@
-{ atomEnv
-, autoPatchelfHook
-, dpkg
-, fetchurl
-, lib
+{ lib
+, buildNpmPackage
+, electron
+, fetchFromGitHub
+, glib
+, gtk3
+, jq
 , libsecret
-, libxshmfence
 , makeDesktopItem
 , makeWrapper
-, stdenv
-, udev
+, moreutils
+, nodejs-16_x
+, pkg-config
+, rustPlatform
 , wrapGAppsHook
 }:
 
-stdenv.mkDerivation rec {
-  pname = "bitwarden";
-  version = "2022.12.0";
+let
+  common = import ./common.nix { inherit lib makeDesktopItem; };
+  buildNpmPackage' = buildNpmPackage.override { nodejs = nodejs-16_x; };
 
-  src = fetchurl {
-    url = "https://github.com/bitwarden/clients/releases/download/desktop-v${version}/Bitwarden-${version}-amd64.deb";
-    sha256 = "sha256-deQG1GEYmnQmO5+jASsiQmDphb+4Jyq9aSn/A4z++uE=";
+  version = "2023.1.1";
+  src = fetchFromGitHub {
+    owner = "bitwarden";
+    repo = "clients";
+    rev = "desktop-v${version}";
+    sha256 = "YEHPDUa0BK8dtaIeWv2kICj6IZIOXUG13mCRzRk80ZY=";
   };
 
-  desktopItem = makeDesktopItem {
-    name = "bitwarden";
-    exec = "bitwarden %U";
-    icon = "bitwarden";
-    comment = "A secure and free password manager for all of your devices";
-    desktopName = "Bitwarden";
-    categories = [ "Utility" ];
+  desktop-native = rustPlatform.buildRustPackage rec {
+    pname = "bitwarden-desktop-native";
+    inherit src version;
+    sourceRoot = "source/apps/desktop/desktop_native";
+    cargoSha256 = "qSpLMYwFtE7BDVcUm7ycpKnfJSvlAUGL2KFoaSjREBM=";
+
+    nativeBuildInputs = [
+      pkg-config
+      wrapGAppsHook
+    ];
+    buildInputs = [
+      glib
+      gtk3
+      libsecret
+    ];
+
+    # Requires /etc/machine-id
+    doCheck = false;
```

Comment:
This can be hacked with libredirect: see https://github.com/NixOS/nixpkgs/pull/212682/files for example.